homeassistant/components/samsungtv/media_player.py

```diff
@@ -141,57 +88,37 @@ def __init__(self, host, port, name, timeout, mac, broadcast, uuid):
         # Generate a configuration for the Samsung library
         self._config = {
             "name": "HomeAssistant",
-            "description": name,
+            "description": self._name,
             "id": "ha.component.samsung",
-            "method": None,
-            "port": port,
-            "host": host,
-            "timeout": timeout,
+            "method": config_entry.data[CONF_METHOD],
+            "port": config_entry.data.get(CONF_PORT),
+            "host": config_entry.data[CONF_HOST],
+            "timeout": 1,
         }
 
-        # Select method by port number, mainly for fallback
-        if self._config["port"] in (8001, 8002):
-            self._config["method"] = "websocket"
-        elif self._config["port"] == 55000:
-            self._config["method"] = "legacy"
+    async def async_added_to_hass(self):
+        """Set up config_entry update listener."""
+        self._update_listener = self._config_entry.add_update_listener(
+            self.update_device_info
+        )
+
+    async def async_will_remove_from_hass(self):
+        """Remove config_entry update listener."""
+        if self._update_listener is not None:
+            self._update_listener()
+
+    async def update_device_info(self, config_entry):
+        """Update device info according to config_entry."""
+        self._uuid = config_entry.data.get(CONF_ID)
+        self._manufacturer = config_entry.data.get(CONF_MANUFACTURER)
+        self._model = config_entry.data.get(CONF_MODEL)
```

Comment:
This is still not working proberly:

```
2020-01-09 20:21:38 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File ".../home-assistant/homeassistant/data_entry_flow.py", line 109, in async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
  File ".../home-assistant/homeassistant/data_entry_flow.py", line 172, in _async_handle_step
    result: Dict = await getattr(flow, method)(user_input)
  File ".../home-assistant/homeassistant/components/samsungtv/config_flow.py", line 168, in async_step_ssdp
    self.hass.config_entries.async_update_entry(config_entry)
  File ".../home-assistant/homeassistant/config_entries.py", line 742, in async_update_entry
    self.hass.async_create_task(listener(self.hass, entry))
TypeError: 'NoneType' object is not callable
```

Reply:
Not sure about the stack trace but if we want to update device info after entity addition we need to use this:
https://github.com/home-assistant/home-assistant/blob/669844e4ddc7c3733ef03e4baf0e3ba6d1c5c3a6/homeassistant/helpers/device_registry.py#L150